# Lecture 1h: Bivariate Rejection Sampling (Algorithm 1.22)
**Simulations in Statistics (52001)**

---

## Overview

We want to sample from a **truncated standard bivariate normal** distribution on the region $\{X_1 \geq a_1,\; X_2 \geq a_2\}$, where $X_1, X_2 \sim N(0,1)$ independently.

**Target density:**

$$f(x_1, x_2) = \frac{1}{2\pi(1 - \Phi(a_1))(1 - \Phi(a_2))}\,e^{-\frac{1}{2}(x_1^2 + x_2^2)}, \quad x_1 \geq a_1,\; x_2 \geq a_2$$

where $\Phi$ is the CDF of the standard normal.

**Proposal density** (product of two shifted exponentials):

$$g(x_1, x_2) = a_1 a_2\,e^{-a_1(x_1 - a_1) - a_2(x_2 - a_2)}, \quad x_1 \geq a_1,\; x_2 \geq a_2$$

**Envelope constant:**

$$c = \max_{x_1 \geq a_1,\, x_2 \geq a_2} \frac{f(x_1, x_2)}{g(x_1, x_2)} = \frac{\exp\!\big\{-\tfrac{1}{2}(a_1^2 + a_2^2)\big\}}{2\pi(1 - \Phi(a_1))(1 - \Phi(a_2))\,a_1\,a_2}$$

**Acceptance ratio** (after simplification):

$$\frac{f(x_1, x_2)}{c \cdot g(x_1, x_2)} = \exp\!\Big\{-\tfrac{1}{2}\big[(x_1 - a_1)^2 + (x_2 - a_2)^2\big]\Big\}$$

**Algorithm 1.22:** Accept row $(X_{1i}, X_{2i})$ if $U_i \leq \exp\!\big\{-\tfrac{1}{2}[(X_{1i} - a_1)^2 + (X_{2i} - a_2)^2]\big\}$.

## R Implementation

In [4]:
acceptance_ratio <- function(x1, x2, a1, a2) {
  exp(-0.5 * ((x1 - a1)^2 + (x2 - a2)^2))
}

Part1h <- function(X, a, U) {
  ratio <- acceptance_ratio(X[, 1], X[, 2], a[1], a[2])
  is_accepted <- U <= ratio
  return(X[is_accepted, , drop = FALSE])
}

In [5]:
X <- matrix(c(
    3.220, 3.536,
    2.463, 3.997,
    2.446, 3.905,
    2.451, 3.773,
    2.937, 3.838,
    2.464, 3.523,
    2.450, 3.567,
    3.161, 4.078,
    2.978, 3.740,
    2.491, 3.648
), ncol = 2, byrow = TRUE)

a <- c(2.4, 3.5)
U <- c(0.7304574, 0.6344974, 0.8087954, 0.7085961, 0.0465277, 0.3968059, 0.3477789, 0.5352538, 0.2922454, 0.986348)

cat("a =", a, "\n")
cat("U =", U, "\n")
cat("X:\n")
print(X)

a = 2.4 3.5 
U = 0.7304574 0.6344974 0.8087954 0.7085961 0.0465277 0.3968059 0.3477789 0.5352538 0.2922454 0.986348 
X:
       [,1]  [,2]
 [1,] 3.220 3.536
 [2,] 2.463 3.997
 [3,] 2.446 3.905
 [4,] 2.451 3.773
 [5,] 2.937 3.838
 [6,] 2.464 3.523
 [7,] 2.450 3.567
 [8,] 3.161 4.078
 [9,] 2.978 3.740
[10,] 2.491 3.648


## Computing the Results

In [6]:
x_accepted          <- Part1h(X, a, U)
accepted_first_x1   <- x_accepted[1, 1]
accepted_first_x2   <- x_accepted[1, 2]
proportion_accepted <- nrow(x_accepted) / nrow(X)

cat("a. X.acc[1,1]            =", accepted_first_x1,   "\n")
cat("b. X.acc[1,2]            =", accepted_first_x2,   "\n")
cat("c. Proportion accepted   =", proportion_accepted,  "\n")

a. X.acc[1,1]            = 2.463 
b. X.acc[1,2]            = 3.997 
c. Proportion accepted   = 0.8 


## Interpretation

- **Accept/reject rule:** Accept row $(X_{1i}, X_{2i})$ if $U_i \leq \exp\!\big\{-\tfrac{1}{2}[(X_{1i} - a_1)^2 + (X_{2i} - a_2)^2]\big\}$
- **X.acc[1,1] and X.acc[1,2]:** the first accepted proposal pair
- **Proportion accepted:** the fraction of proposal rows that passed the acceptance test

## Final Answer

Accept $(X_{1i}, X_{2i})$ if $U_i \leq \exp\!\big\{-\tfrac{1}{2}[(X_{1i} - a_1)^2 + (X_{2i} - a_2)^2]\big\}$

$$f(x_1, x_2) = \frac{1}{2\pi(1 - \Phi(a_1))(1 - \Phi(a_2))}\,e^{-\frac{1}{2}(x_1^2 + x_2^2)}, \quad a = (3.2,\; 3.5)$$

$$\text{(a)}\ X.\text{acc}[1,1] \qquad \text{(b)}\ X.\text{acc}[1,2] \qquad \text{(c)}\ \frac{|\text{accepted rows}|}{|\text{proposal rows}|}$$